### PR TITLE
LA-368 Duplicate in Shared Spaces

### PR DESCRIPTION
**Pull request by imGok**
duplicate

### CODE REVIEW

File: domain/lib/src/model/sharedspace/shared_space_operation_role.dart [hunk 1]
```diff
@@ -61,8 +61,20 @@ class SharedSpaceOperationRole {
...
     SharedSpaceRoleName.WRITER
   ];
+
+  static const duplicateNodeSharedSpaceRoles = [
+    SharedSpaceRoleName.ADMIN,
+    SharedSpaceRoleName.WRITER,
+    SharedSpaceRoleName.CONTRIBUTOR
+  ];
+
+  static const renameNodeSharedSpaceRoles = [
```
Comment (hoangdat) [inline domain/lib/src/model/sharedspace/shared_space_operation_role.dart:75]: Why you need these roles?

File: lib/presentation/widget/shared_space_document/shared_space_document_widget.dart [hunk 2]
```diff
@@ -386,15 +386,16 @@ class _SharedSpaceDocumentWidgetState extends State<SharedSpaceDocumentWidget> {
               context,
               workGroupNodes,
               _moreActionList(context, documents),
-              _removeWorkGroupNodeAction(workGroupNodes, itemSelectionType: ItemSelectionType.multiple)))
+              SharedSpaceOperationRole.deleteNodeSharedSpaceRoles.contains(_arguments.sharedSpaceNode.sharedSpaceRole.name) ? _removeWorkGroupNodeAction(workGroupNodes, itemSelectionType: ItemSelectionType.multiple) : SizedBox.shrink()))
```
Comment (hoangdat) [inline lib/presentation/widget/shared_space_document/shared_space_document_widget.dart:389]: too long line, please split it
```
SharedSpaceOperationRole.deleteNodeSharedSpaceRoles.contains(_arguments.sharedSpaceNode.sharedSpaceRole.name) 
  ? _removeWorkGroupNodeAction(workGroupNodes, itemSelectionType: ItemSelectionType.multiple) 
  : SizedBox.shrink()))
```

File: lib/presentation/widget/shared_space_document/shared_space_document_widget.dart [hunk 3]
```diff
@@ -386,15 +386,16 @@ class _SharedSpaceDocumentWidgetState extends State<SharedSpaceDocumentWidget> {
...
           .build();
   }
 
   List<Widget> _moreActionList(BuildContext context, List<WorkGroupNode> workGroupNodes) {
     return [
       if (Platform.isIOS) _exportFileAction(workGroupNodes, itemSelectionType: ItemSelectionType.multiple),
       if (Platform.isAndroid) _downloadFilesAction(workGroupNodes, itemSelectionType: ItemSelectionType.multiple),
-      _copyToAction(context, workGroupNodes, itemSelectionType: ItemSelectionType.multiple)
+      _copyToAction(context, workGroupNodes, itemSelectionType: ItemSelectionType.multiple),
+      if (SharedSpaceOperationRole.duplicateNodeSharedSpaceRoles.contains(_arguments.sharedSpaceNode.sharedSpaceRole.name)) _duplicateMultipleSelection(workGroupNodes),
```
Comment (hoangdat) [inline lib/presentation/widget/shared_space_document/shared_space_document_widget.dart:398]: why not 
```
if (SharedSpaceOperationRole.duplicateNodeSharedSpaceRoles.contains(_arguments.sharedSpaceNode.sharedSpaceRole.name)) {
  _duplicateMultipleSelection(workGroupNodes),
}
```

Comment (hoangdat) [reply lib/presentation/widget/shared_space_document/shared_space_document_widget.dart:398]: same for other

File: domain/lib/src/model/sharedspace/shared_space_operation_role.dart (hunk 1 above)
Comment (imGok) [reply domain/lib/src/model/sharedspace/shared_space_operation_role.dart:75]: I'm referring to the permissions document and I saw it was missing so I added it

File: lib/presentation/widget/shared_space_document/shared_space_document_widget.dart (hunk 3 above)
Comment (imGok) [reply lib/presentation/widget/shared_space_document/shared_space_document_widget.dart:398]: I don't think it's possible in this case, dart recognizes it as a collection when I add brackets

Comment (imGok) [reply lib/presentation/widget/shared_space_document/shared_space_document_widget.dart:398]: Don't think it acts as a normal if else statement

**Review by hoangdat — APPROVED** [review]

**Pull request merged**